tools/kms_secrets_encryption.py

```diff
@@ -0,0 +1,548 @@
+        parent   = self.client.location_path(self.project_id, self.location)
+        response = self.client.create_key_ring(parent, key_ring_id, keyring)
+
+        return response.name
+
+
+    def decrypt_ciphertext(self, ciphertext):
+        """A method that decrypts ciphertext.
+
+        Uses GCP KMS to decrypt ciphertext back to plaintext using the provided
+        symmetric crypto key that belongs to this instance.
+        
+        Args:
+            ciphertext (str): the ciphertext being decrypted
+        Returns:
+            string: the plaintext
+        """
+
+        # Convert ciphertext string to a byte string
+        ciphertext = ciphertext.encode("utf-8")
+
+        # Base64 decoding of ciphertext
+        ciphertext = base64.b64decode(ciphertext)
+
+        # Use the KMS API to decrypt the data.
+        response = self.client.decrypt(self.crypto_key_path, ciphertext)
+
+        # Decode Base64 plaintext
+        plaintext = response.plaintext.decode()
+        return plaintext
+
+
+    def decrypt_file(self, file_path):
+        """A method that decrypts the contents of a text file.
+
+        Uses GCP KMS to decrypt ciphertext back to plaintext using the provided
+        symmetric crypto key that belongs to this instance.
+        
+        Args:
+            file_path (str): the path of the text file being decrypted
+        Returns:
+            string: the path to the decrypted text file created
+        """
+
+        try:
+            print("Decrypting file: {}...".format(file_path))
+
+            with open(file_path) as f:
+                file_ciphertext = f.read()
+        
+            file_decrypted = self.decrypt_ciphertext(file_ciphertext)
+
+            file_path_decrypted = "{}".format(file_path).replace(".encrypted", "")
+
+            with open(file_path_decrypted, "w") as f:
+                f.write(file_decrypted)
+
+            os.remove(file_path)
+
+        except Exception as err:
+            print("An exception occurred decrypting file.")
+            print("{}\n".format(err))
+            raise SystemExit()
+        
+        return file_path_decrypted
+
+
+    def decrypt_tfvars_secrets(self):
+        """A method that decrypts the secrets contained in the terraform.tfvars file.
+
+        This method contains the logic for handling the decryption of the secrets 
+        and any file paths associated with it using GCP KMS. Once decrypted, it calls 
+        write_new_tfvars() to write all secrets to a new terraform.tfvars file. 
+        """    
+
+        # Abort the decryption if the tfvars is in plaintext using kms_cryptokey_id as an indicator
+        if not self.tfvars_data.get("kms_cryptokey_id"):
+            print("No kms_cryptokey_id present in tfvars. Ensure the secrets are encrypted and try again.\n")
+            raise SystemExit()
+
+        # Abort the decryption if GCP credentials is missing
+        if not self.tfvars_data.get("gcp_credentials_file"):
```

Comment:
could move this check out to the parent since both encrypt and decrypt function calls this.